src/ExtCmd.actor.cpp

```diff
@@ -406,6 +406,80 @@ struct DropCollectionCmd {
 };
 REGISTER_CMD(DropCollectionCmd, "drop");
 
+ACTOR static Future<int> internal_doRenameCollectionIndexesActor(Reference<DocTransaction> tr,
+                                                                 Namespace ns,
+                                                                 Reference<MetadataManager> mm) {
+	state Reference<UnboundCollectionContext> unbound = wait(mm->getUnboundCollectionContext(tr, ns));
+	unbound->bindCollectionContext(tr)->bumpMetadataVersion();
+	TraceEvent(SevInfo, "BumpMetadataVersion")
+	    .detail("reason", "renameCollection")
+	    .detail("ns", fullCollNameToString(ns));
+	return true;
+}
+
+ACTOR static Future<Void> Internal_doRenameCollection(Reference<DocTransaction> tr,
+                                                      Reference<ExtMsgQuery> query,
+                                                      Reference<ExtConnection> ec) {
+	state Namespace ns;
+	ns.first = upOneLevel(query->ns.second);
+	state std::string sourceCollection = getLastPart(query->ns.second);
+	state std::string destinationCollection = getLastPart(query->query.getStringField("to"));
+	state bool dropTarget = query->query.getBoolField("dropTarget");
+
+	state Future<bool> exists_destinationCollectionF =
+	    ec->docLayer->rootDirectory->exists(tr->tr, {StringRef(ns.first), StringRef(destinationCollection)});
+	state Future<bool> exists_sourceCollectionF =
+	    ec->docLayer->rootDirectory->exists(tr->tr, {StringRef(ns.first), StringRef(sourceCollection)});
+	wait(success(exists_destinationCollectionF) && success(exists_sourceCollectionF));
+
+	if (exists_sourceCollectionF.get()) {
+		if (exists_destinationCollectionF.get()) {
+			if (dropTarget) {
+				ns.second = destinationCollection;
+				state Reference<UnboundCollectionContext> unbound = wait(ec->mm->getUnboundCollectionContext(tr, ns));
+				wait(success(internal_doDropIndexesActor(tr, ns, ec->mm)));
+				wait(unbound->collectionDirectory->remove(tr->tr));
+			}
+		}
+	} else {
+		throw directory_does_not_exist();
+	}
+
+	Reference<DirectorySubspace> Collection =
+	    wait(ec->docLayer->rootDirectory->move(tr->tr, {StringRef(ns.first), StringRef(sourceCollection)},
+	                                           {StringRef(ns.first), StringRef(destinationCollection)}));
+	ns.second = sourceCollection;
```

Comment:
Indexes are maintained in `system.indexes`. Seems like we are not changing it. So, even after rename indexes point to old collection.